code/app/src/main/java/com/example/snackoverflow/AddRecipe.java

```diff
@@ -0,0 +1,49 @@
+package com.example.snackoverflow;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import android.content.Intent;
+
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.View;
+
+import com.github.dhaval2404.imagepicker.ImagePicker;
+import de.hdodenhof.circleimageview.CircleImageView;
+
+// TODO: How to request user permission for gallery access with the new
+// Android API
+public class AddRecipe extends AppCompatActivity {
+
+    public CircleImageView imageView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_recipe);
+        imageView = findViewById(R.id.addPhoto);
+
+        imageView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ImagePicker.with(AddRecipe.this)
+                        .crop()
+                        .maxResultSize(1080, 1080)    //Final image resolution will be less than 1080 x 1080(Optional)
+                        .start(20);
+
+            }
+        });
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
```

Comment:
Double check. I think onActivityResult is deprecated.

Reply:
The onActivityResult method as it is, is not deprecated. But it makes a super call which is deprecated. Try avoiding it

<img width="762" alt="image" src="https://user-images.githubusercontent.com/69052109/197643572-5d52b5dc-5464-47b6-9110-89e6e79479ee.png">